git/git.go

```diff
@@ -37,11 +37,21 @@ func Dir() (string, error) {
 }
 
 func HasFile(segments ...string) bool {
-	dir, err := Dir()
-	if err != nil {
-		return false
+	// The blessed way to resolve paths in git dir, since Git 1.5.0
+	// Needed when working with git worktrees.
+	output, err1 := gitOutput("rev-parse", "-q", "--git-path", filepath.Join(segments...))
```

Comment:
It seems to me that this logic belongs to `Dir()` function instead of `HasFile`. The `Dir()` function is supposed to return the path to the `.git` directory, right?